src/data_source/storage/sql/queries/node.rs

```diff
@@ -340,9 +341,18 @@ impl<Types: NodeType> UpdateAggregatesStorage<Types> for Transaction<Write> {
             "aggregate",
             ["height", "num_transactions", "payload_size"],
             ["height"],
-            rows,
+            rows.clone(),
         )
-        .await
+        .await?;
+
+        let (height, num_transactions, payload_size) =
+            rows.last().ok_or_else(|| anyhow!("no row"))?;
```

Comment:
Nit: I think `.ok_or_else(|| anyhow!(whatever))` is more simply written `.context(whatever)`

Reply:
yeah, this had conflict with snafu context